client/task_runner.go

```diff
@@ -401,7 +402,7 @@ func (r *TaskRunner) run() {
 				close(stopCollection)
 
 				// Store that the task has been destroyed and any associated error.
-				r.setState(structs.TaskStateDead, structs.NewTaskEvent(structs.TaskKilled).SetKillError(err))
+				r.setState(structs.TaskStateDead, r.destroyEvent.SetKillError(err))
```

Comment:
So I think this is somewhat out of scope but my issue with this is that we want to store the `TaskKilled` event after the task dies and we also want to store why. The issue is that we are conflating `TaskKilled` with the user sending that signal and the task being killed.

So for now I would revert 405 to what it was and add a new one `r.setState(structs.TaskStateDead, r.destroyEvent)`. In a separate PR we can add a new event called `TaskUserStopped`